client/app/pages/queries-list/index.js

```diff
@@ -43,16 +80,49 @@ class QueriesListCtrl {
       });
     }
 
-    this.paginator = new LivePaginator(queriesFetcher, { page });
+    this.paginator = new LivePaginator(
+      queriesFetcher,
+      {
+        page,
+        itemsPerPage: this.pageSize,
+        orderByField: pageOrder,
+        orderByReverse: pageOrderReverse,
+        params: this.parameters,
+      },
+    );
+
+    this.parameters = () => ({
+      searchTerm: this.searchTerm,
+      showDrafts: this.showDrafts,
+    });
 
     this.tabs = [
       { path: 'queries', name: 'All Queries', isActive: path => path === '/queries' },
       { name: 'My Queries', path: 'queries/my' },
-      { name: 'Search', path: 'queries/search' },
     ];
 
-    this.showList = () => this.paginator.getPageRows() !== undefined && this.paginator.getPageRows().length > 0;
-    this.showEmptyState = () => this.paginator.getPageRows() !== undefined && this.paginator.getPageRows().length === 0;
+    this.searchUsed = () => this.searchTerm !== undefined || this.searchTerm !== '';
+
+    this.hasResults = () => this.paginator.getPageRows() !== undefined &&
+      this.paginator.getPageRows().length > 0;
+
+    this.showEmptyState = () => !this.hasResults() && !this.searchUsed();
+
+    this.showDraftsCheckbox = () => $location.path() !== '/queries/my';
+
+    this.clearSearch = () => {
+      this.searchTerm = '';
+      this.update();
+    };
+
+    this.update = () => {
+      if (!isString(this.searchTerm) || this.searchTerm.trim() === '') {
+        this.searchTerm = '';
+      }
+      this.paginator.itemsPerPage = this.pageSize;
+      this.paginator.params = this.parameters();
```

Comment:
any reason not to just inline the `parameters` function? I found the similar name to `params` confusing.

Reply:
Yeah, good idea.